src/shared/decorators/future-dates-not-allowed.decorator.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  ValidationArguments,
+  ValidationOptions,
+  registerDecorator,
+} from 'class-validator';
+
+export function IsNotFutureDate(validationOptions?: ValidationOptions) {
+  return function (object: object, propertyName: string) {
+    registerDecorator({
+      name: 'isNotFutureDate',
+      target: object.constructor,
+      propertyName: propertyName,
+      options: validationOptions,
+      validator: {
```

Comment:
The function doesn't check if the date is valid, possible returning an error when trying to compare an invalid date with a valid one.
The following suggestion checks that and returns false if invalid using isNaN

validator: {
  validate(value: any) {
    const currentDate = new Date();
    const inputDate = new Date(value);

    if (isNaN(inputDate.getTime())) {
      return false;
    }

    return inputDate <= currentDate;
  },
  defaultMessage(args: ValidationArguments) {
    return `${args.property} should not be a future date and a valid one`;
  },
},

Reply:
Nice! And done.